userland/examples/ble_adv/ble.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <tock.h>
+
+/*------- SYSCALLS------------*/
+#define DRIVER_RADIO      33
+
+// commands calls
+#define BLE_ADV_START     0
+#define BLE_ADV_STOP      1
+#define CFG_TX_POWER      2
+#define CFG_ADV_INTERVAL  3
+/*----------------------------*/
+
+// AD Types
+#define BLE_HS_ADV_TYPE_FLAGS                   0x01
+#define BLE_HS_ADV_TYPE_INCOMP_UUIDS16          0x02
+#define BLE_HS_ADV_TYPE_COMP_UUIDS16            0x03
+#define BLE_HS_ADV_TYPE_INCOMP_UUIDS32          0x04
+#define BLE_HS_ADV_TYPE_COMP_UUIDS32            0x05
+#define BLE_HS_ADV_TYPE_INCOMP_UUIDS128         0x06
+#define BLE_HS_ADV_TYPE_COMP_UUIDS128           0x07
+#define BLE_HS_ADV_TYPE_INCOMP_NAME             0x08
+#define BLE_HS_ADV_TYPE_COMP_NAME               0x09
+#define BLE_HS_ADV_TYPE_TX_PWR_LVL              0x0a
+#define BLE_HS_ADV_TYPE_SLAVE_ITVL_RANGE        0x12
+#define BLE_HS_ADV_TYPE_SOL_UUIDS16             0x14
+#define BLE_HS_ADV_TYPE_SOL_UUIDS128            0x15
+#define BLE_HS_ADV_TYPE_SVC_DATA_UUID16         0x16
+#define BLE_HS_ADV_TYPE_PUBLIC_TGT_ADDR         0x17
+#define BLE_HS_ADV_TYPE_RANDOM_TGT_ADDR         0x18
+#define BLE_HS_ADV_TYPE_APPEARANCE              0x19
+#define BLE_HS_ADV_TYPE_ADV_ITVL                0x1a
+#define BLE_HS_ADV_TYPE_SVC_DATA_UUID32         0x20
+#define BLE_HS_ADV_TYPE_SVC_DATA_UUID128        0x21
+#define BLE_HS_ADV_TYPE_URI                     0x24
+#define BLE_HS_ADV_TYPE_MFG_DATA                0xff
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int ble_adv_data(uint8_t type, uint8_t len, const unsigned char *data);
+int ble_adv_set_txpower(uint8_t);
```

Comment:
Yes and no. You're correct that the data is still 8 bits regardless and can be interpreted in any way. However, as a user of the interface, if I didn't know how to use it and looked at the header file, I would assume it only takes positive numbers because it's type is unsigned. You do have defines for valid TX powers, but you didn't enforce them as an enum, so it seems like the interface _could_ take any value.

My opinion, is that you should either allow it to accept signed numbers (-30, -20, ... -4, 0, 4) in which case you will have to deal with the user possibly giving you an invalid choice (100 or -7, etc.) or change its interface to accept an enum which would force the user to only give you valid options:

```
typedef enum {
    ...
    TX_POWER_NEG_20_DBM = -20,
    TX_POWER_NEG_30_DBM = -30,
} BLE_TxPower_t;

int ble_adv_set_txpower(BLE_TxPower_t power);
```